### ISSUE TITLE
some observations / enhancement request

### ISSUE THREAD

**Issue by argupta23** [strongloop/loopback4-example-kafka#3]
@raymondfeng 

Thanks for sharing this example.

I tried using your example code and have a few observations that I would like to share with you.

1.  I think we should make "offset" a user input.
2.  For the case when input "limit" is greater than the number of messages within the selected topic, the explorer is stuck in "Loading" state. I can see that it gets the messages from the backend, but since the messages are less than the requested limit, it just sits there for a timeout to occur and eventually error's out.

Please let me know your thoughts. 

Based on your feedback, I am looking forward to add a few things to this example and will be happy to share an updated version with the community.

BTW I am using the latest versions of the following packages.
    "@loopback/authentication": "^5.0.0",
    "@loopback/authentication-jwt": "^0.5.0",
    "@loopback/authorization": "^0.6.4",
    "@loopback/boot": "^2.4.1",
    "@loopback/core": "^2.9.3",
    "@loopback/extension-health": "^0.4.11",
    "@loopback/extension-metrics": "^0.3.11",
    "@loopback/repository": "^2.11.0",
    "@loopback/rest": "^6.0.0",
    "@loopback/rest-crud": "^0.8.11",
    "@loopback/rest-explorer": "^2.2.8",
    "@loopback/service-proxy": "^2.3.6",
    "kafka-node": "^5.0.0",

**Comment by raymondfeng** [comment]
@argupta23 Maybe you should checkout https://github.com/spirAde/loopback4-kafkajs-extension

**Comment by argupta23** [comment]
@raymondfeng 

Thanks for sharing. It looks interesting but is still in the very initial stages.

Pls. don't get me wrong. One of the concerns with extensions is that many a times they are not updated, so you are kind of stuck. I will continue to keep an eye on that package.

One of the things I want to do is to add support for socketio to your current example. I will try to take a stab at it over the weekend and will reachout to you for help.

If you already have something along the lines of Kafka + Socketio integrated together, I would he happy to use that and not start from scratch.

Thanks

**Comment by raymondfeng** [comment]
@argupta23 I agree that your scenario is not uncommon. We have been exploring possible LoopBack extensions to support the messaging related use cases.

- https://github.com/strongloop/loopback-next/pull/6071
- https://github.com/strongloop/loopback-next/tree/pubsub/packages/pubsub

Unfortunately the progress has been slow due to our constrained bandwidth.

**Comment by argupta23** [comment]
@raymondfeng 

Appreciate the pointers I will look into them. I am willing to help you folks out by integrating some of the many silos that exist today.

Please let me know how I can help the team out.

Relative to the initial ask, I have already taken care of point 1.
Can you please share some thoughts on how one would deal with point 2?

Thanks
